tensorflow/python/data/ops/dataset_ops.py

```diff
@@ -3229,6 +3229,61 @@ def select_dataset_varying_logits(logits, seed):
     return _DirectedInterleaveDataset(selector_input, datasets,
                                       stop_on_empty_dataset)
 
+  @staticmethod
+  def choose_from_datasets(datasets,
+                           choice_dataset,
+                           stop_on_empty_dataset=False):
+    """Creates a dataset that deterministically chooses elements from `datasets`.
+
+    For example, given the following datasets:
+
+    ```python
+    datasets = [tf.data.Dataset.from_tensors("foo").repeat(),
+                tf.data.Dataset.from_tensors("bar").repeat(),
+                tf.data.Dataset.from_tensors("baz").repeat()]
+
+    # Define a dataset containing `[0, 1, 2, 0, 1, 2, 0, 1, 2]`.
+    choice_dataset = tf.data.Dataset.range(3).repeat(3)
+
+    result = tf.data.Dataset.choose_from_datasets(datasets, choice_dataset)
+    ```
+
+    The elements of `result` will be:
+
+    ```
+    "foo", "bar", "baz", "foo", "bar", "baz", "foo", "bar", "baz"
+    ```
+
+    Args:
+      datasets: A non-empty list of `tf.data.Dataset` objects with compatible
+        structure.
+      choice_dataset: A `tf.data.Dataset` of scalar `tf.int64` tensors
+        between `0` and `len(datasets) - 1`.
+      stop_on_empty_dataset: If `True`, selection stops if it encounters an
+        empty dataset. If `False`, it skips empty datasets. It is recommended
+        to set it to `True`. Otherwise, the selected elements start off as
+        the user intends, but may change as input datasets become empty.
+        This can be difficult to detect since the dataset starts off looking
+        correct. Default to `False` for backward compatibility.
```

Comment:
Can we make the default `True` now? The backward compatibility requirement no longer applies since we are migrating the API. We need to call out the change in default in the release notes and deprecation message, and tell users to set `stop_on_empty_dataset=False` to get the same as the experimental API.

We should do the same for `sample_from_datasets`, in a separate CL.

Reply:
@aaudiber  I agree. I have made the necessary changes. Please take a look.